sdk/typescript/src/module/introspector/typescript_module/ast.ts

```diff
@@ -407,6 +407,65 @@ export class AST {
     }
   }
 
+  // Try to extract literal values (enum members, string/number/boolean/bigint
+  // literals) directly from the type checker. Returns undefined if the
+  // expression is not a literal.
+  private getLiteralValueFromExpression(
+    expression: ts.Expression,
+  ): string | number | boolean | bigint | undefined {
+    const type = this.checker.getTypeAtLocation(expression)
+    if (!type) {
+      return undefined
+    }
+
+    const resolveLiteral = (
+      t: ts.Type,
+    ): string | number | boolean | bigint | undefined => {
+      if (t.flags & ts.TypeFlags.BooleanLiteral) {
+        const intrinsic = t as unknown as { intrinsicName?: string }
+        if (intrinsic.intrinsicName === "true") {
+          return true
+        }
+        if (intrinsic.intrinsicName === "false") {
+          return false
+        }
+      }
+
+      if (
+        t.flags &
+        (ts.TypeFlags.EnumLiteral |
+          ts.TypeFlags.StringLiteral |
+          ts.TypeFlags.NumberLiteral |
+          ts.TypeFlags.BigIntLiteral)
+      ) {
+        const literal = t as ts.LiteralType
+        if (literal.value !== undefined) {
+          return literal.value as string | number | bigint
+        }
+
+        const intrinsic = literal as unknown as { intrinsicName?: string }
+        if (intrinsic.intrinsicName !== undefined) {
+          return intrinsic.intrinsicName as string
+        }
+      }
+
+      return undefined
+    }
+
+    if (type.isUnion()) {
+      for (const subtype of type.types) {
+        const literal = resolveLiteral(subtype)
+        if (literal !== undefined) {
+          return literal
+        }
+      }
```

Comment:
I'm not sure about that part, does this mean we can handle union like `"foo" | "bar" ????`
I'm not sure the typedef introspection handle that